components/script/dom/xrview.rs

```diff
@@ -90,14 +88,18 @@ impl XRViewMethods for XRView {
     }
 
     /// https://immersive-web.github.io/webxr/#dom-xrview-projectionmatrix
-    fn ProjectionMatrix(&self, _cx: JSContext) -> NonNull<JSObject> {
-        if self.proj.get().is_null() {
+    fn ProjectionMatrix(&self, _cx: JSContext) -> Float32Array {
+        if !self.proj.is_initialized() {
             let cx = GlobalScope::get_cx();
             // row_major since euclid uses row vectors
             let proj = self.view.projection.to_array();
-            create_typed_array(cx, &proj, &self.proj);
+            let _ = self.proj.set_data(cx, &proj);
+        }
+
+        match self.proj.get_internal() {
+            Ok(array) => array,
+            Err(_) => unreachable!(),
```

Comment:
are we should this is unreachable ?

Reply:
AFAIK, If we handle the error from set_data (initialization fails), then L101 is unreachable.

Reply:
Since the [WebIDL](https://immersive-web.github.io/webxr/#ref-for-dom-xrview-projectionmatrix%E2%91%A1) and the [backing algorithm in the spec](https://immersive-web.github.io/webxr/#xrview-obtain-the-projection-matrix)do not allow for failure, there is no way we can handle failure, so one could simply `expect` on both errors.